java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java

```diff
@@ -0,0 +1,342 @@
+  private SeekBar mMaxDistanceSeekBar;
+  private TextView mCurrHeading;
+  private TextView mCurrPitch;
+  private TextView mCurrHorizontalAngle;
+  private TextView mCurrVerticalAngle;
+  private TextView mCurrMinDistance;
+  private TextView mCurrMaxDistance;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // set initial values
+    mMinDistance = mInitMinDistance;
+    mMaxDistance = mInitMaxDistance;
+
+    // create a scene and add an imagery basemap to it
+    mSceneView = findViewById(R.id.sceneView);
+    ArcGISScene scene = new ArcGISScene();
+    scene.setBasemap(Basemap.createImagery());
+    mSceneView.setScene(scene);
+
+    // add base surface for elevation data
+    Surface surface = new Surface();
+    final String localElevationImageService = getString(
+        com.esri.arcgisruntime.sample.viewshedlocation.R.string.elevation_service);
+    surface.getElevationSources().add(new ArcGISTiledElevationSource(localElevationImageService));
+    scene.setBaseSurface(surface);
+
+    // add a scene layer
+    final String buildings = getString(com.esri.arcgisruntime.sample.viewshedlocation.R.string.buildings_layer);
+    ArcGISSceneLayer sceneLayer = new ArcGISSceneLayer(buildings);
+    scene.getOperationalLayers().add(sceneLayer);
+
+    // create viewshed from location
+    Point location = new Point(-4.50, 48.4, 100.0);
+    Viewshed.setFrustumOutlineColor(Color.BLUE);
+    mViewshed = new LocationViewshed(location, mInitHeading, mInitPitch, mInitHorizontalAngle, mInitVerticalAngle,
+        mInitMinDistance, mInitMaxDistance);
+    mViewshed.setFrustumOutlineVisible(true);
+
+    // add a camera and set it to orbit the location point
+    Camera camera = new Camera(location, 20000000, 0, 55, 0);
+    OrbitLocationCameraController orbitCamera = new OrbitLocationCameraController(location, 5000);
+    mSceneView.setCameraController(orbitCamera);
+    mSceneView.setViewpointCamera(camera);
+
+    // create an analysis overlay to add the mViewshed to the scene view
+    AnalysisOverlay analysisOverlay = new AnalysisOverlay();
+    analysisOverlay.getAnalyses().add(mViewshed);
+    mSceneView.getAnalysisOverlays().add(analysisOverlay);
+
+    handleUiElements();
+  }
+
+  /**
+   * Handles double touch drag for movement of viewshed location point, inflation of UI elements, and listeners for
+   * changes in seek bar progress.
+   */
+  private void handleUiElements() {
+
+    mSceneView.setOnTouchListener(new DefaultSceneViewOnTouchListener(mSceneView) {
+
+      @Override public boolean onDoubleTouchDrag(MotionEvent motionEvent) {
+
+        // convert from screen point to surface point
+        android.graphics.Point screenPoint = new android.graphics.Point(Math.round(motionEvent.getX()),
+            Math.round(motionEvent.getY()));
+        Point surfacePoint = mSceneView.screenToBaseSurface(screenPoint);
```

Comment:
Specific reason why this is using a slightly different code path to the equivalent JavaSE sample?

Reply:
It makes a noticeable performance hit during the drag (being async), but I suppose .screenToLocationAsync(...) is technically more accurate given this is a viewshed sample.